bokehjs/src/lib/models/annotations/scale_bar.ts

```diff
@@ -552,25 +620,29 @@ export class ScaleBar extends Annotation {
       ["background_", mixins.Hatch],
     ])
 
-    this.define<ScaleBar.Props>(({NonNegative, Float, Str, Ref, Or, Auto}) => ({
-      range:          [ Or(Ref(Range), Auto), "auto" ],
-      unit:           [ Str, "m" ],
-      dimensional:    [ Ref(Dimensional), () => new MetricLength() ],
-      orientation:    [ Orientation, "horizontal" ],
-      bar_length:     [ NonNegative(Float), 0.2 ],
-      length_sizing:  [ LengthSizing, "adaptive" ],
-      location:       [ Anchor, "top_right" ],
-      label:          [ Str, "@{value} @{unit}" ],
-      label_align:    [ Align, "center" ],
-      label_location: [ Location, "below" ],
-      label_standoff: [ Float, 5 ],
-      title:          [ Str, "" ],
-      title_align:    [ Align, "center" ],
-      title_location: [ Location, "above" ],
-      title_standoff: [ Float, 5 ],
-      margin:         [ Float, 10 ],
-      padding:        [ Float, 10 ],
-      ticker:         [ Ref(Ticker), () => new FixedTicker({ticks: []}) ],
+    this.define<ScaleBar.Props>(({NonNegative, Float, Str, Ref, Or, Auto, Enum}) => ({
+      range:            [ Or(Ref(Range), Auto), "auto" ],
+      unit:             [ Str, "m" ],
+      dimensional:      [ Ref(Dimensional), () => new MetricLength() ],
+      orientation:      [ Orientation, "horizontal" ],
+      bar_length:       [ NonNegative(Float), 0.2 ],
+      bar_length_units: [ Enum("screen", "data"), "screen" ],
+      length_sizing:    [ LengthSizing, "adaptive" ],
+      location:         [ Position, "top_right" ],
+      x_units:          [ PositionUnits, "data" ],
+      y_units:          [ PositionUnits, "data" ],
+      anchor:           [ AutoAnchor, "auto" ],
+      label:            [ Str, "@{value} @{unit}" ],
+      label_align:      [ Align, "center" ],
+      label_location:   [ Location, "below" ],
+      label_standoff:   [ Float, 5 ],
+      title:            [ Str, "" ],
+      title_align:      [ Align, "center" ],
+      title_location:   [ Location, "above" ],
+      title_standoff:   [ Float, 5 ],
+      margin:           [ Float, 10 ],
+      padding:          [ Float, 10 ],
+      ticker:           [ Ref(Ticker), () => new FixedTicker({ticks: []}) ],
```

Comment:
can we alphabetize this and make a policy to always alphabetize these lists going forward?

Reply:
On one hand I sympathize, because often there is no structure to these properties. However, sometimes there is at least some structure, like it makes sense to group `x_units` and `y_units` together. Though in this case they will very likely be grouped together anyway, and perhaps it would make even more sense to have a single property for this, i.e. `location_units`. I also try to put important properties at the top, as much as it's possible. That could be achieved with user sortable API docs.